### DISCUSSION TITLE
How to debug event glitch?

### DISCUSSION THREAD

**Discussion by ChibaPet** [azerothcore/azerothcore-wotlk#12421] [General]
Two weeks in a row now my kid has fished up 40 Speckled Tastyfish for the Stranglethorn Fishing Extravaganza, but Riggle Bassbait tells her the "We already have all our winners" text, even though no one else has turned it in. I then got the 40 on one of my characters and he gives me the same text, but he'll let me turn in the quest anyway. My daughter doesn't get the turn-in option.

As much as seeing this fixed, I'd like to know how to go about debugging it. Presumably the event state exists in the database, but I'm not finding where, and I'd love pointers. I followed through https://github.com/azerothcore/azerothcore-wotlk/issues/6973 and that led me to GameEventMgr.cpp, and thence to GameEventMgr::SaveWorldEventStateToDB. That suggests I look at CHAR_INS_GAME_EVENT_SAVE and I see that wanting to update acore_characters.game_event_save, but that's empty.

I started using AzerothCore in February so the related changes and database ought to have existed. I'm confused about how the event can be active but the NPC emits a message suggesting it's completed, and how two different characters can get different responses if the event state is saved in one place. Different players/characters have won the competition for the last several weeks so it *has* worked, but last week and this things seem to be a bit strange.

For kicks, I just rebuilt using source from this morning, but it didn't make a difference.

Thanks in advance for clues.

**Comment by virgo77** [comment]
Hi!
Maybe it's obvious what I will say here. So, sorry for that. I'm just trying to give you some help ;)

To debug the game you need to compile the core in Debug mode or RelWithDebInfo mode. You will have PDB files that can help for debugging.
After that, check the 'AC Wiki' wich explains what to do for debugging? You will need to collect some informations, then pull them here if you can't find any solution :)

Don't forget to check this topic: https://classic.wowhead.com/guides/wow-classic-stranglethorn-vale-fishing-extravaganza
maybe you miss a prerequisite, not even in Fishing as you need to be at last 150 to start the event, but maybe something else

Good luck

**Comment by ChibaPet** [comment]
I appreciate your replying, but I'm struggling to find debugging information on the wiki. Where is it located, there? I see something about logging crashes, but this isn't a crash. This is more my asking about the design of events so I can trace where this is going wrong. I'd like to understand how to observe the database state that tracks event status, assuming that is committed to the database - how else would the correct state be maintained through restarts, for instance?

As noted, I read through https://github.com/azerothcore/azerothcore-wotlk/issues/6973 and from there I read GameEventMgr.cpp and then read the CHAR_INS_GAME_EVENT_SAVE query which got me to look at acore_characters.game_event_save only to find that it's empty. Is it expected that that table shouldn't be populated? Is this state that's only saved in the server? Is it only written to the database on a controlled shutdown or something?

Looking at the Internet for this PDB file you're describing, it seems to be something related to Visual Studio... I'm hoping and assuming it's possible to do useful debugging on a GNU/Linux system. My server is built with RelWithDebInfo but I don't see any .pdb files amidst the build artifacts.

I do wonder if my database is slowly drifting. I don't build in place, I build and then ship the binaries and SQL data to the server that runs the stuff, and it seems as though a bunch of this might make the unsafe assumption that it's all being run in place. My builds are essentially 'git pull', build, make install to a local tree, rsync -av --delete azerothcore-wotlk/data/sql into the data folder in that tree, and then ship the whole local install to the server where it'll run, where presumably the automatic database updater runs and finds everything that's changed since the last checkpoint it knows about. But I'm making a fair number of assumptions here and I'm not finding some of the debugging data you're suggesting, so I'd be grateful for additional pointers.

**Reply by Kitzunu** [reply]
Just going to point to some links tho about debugging
https://www.azerothcore.org/wiki/run-worldserver-and-authserver-in-visual-studio
https://www.azerothcore.org/wiki/how-to-restart-and-debug

**Comment by ChibaPet** [comment]
So, looking at the build, it seems the database updater hardcodes the source directory path, which obviously won't exist when the build and server live on two different systems. This is an error.

-DCMAKE_INSTALL_PREFIX=/opt/acore/ should be honored for sql updates.

It looks like TrinityCore does better, comparing https://github.com/TrinityCore/TrinityCore/blob/master/src/server/database/Updater/DBUpdater.cpp and https://github.com/azerothcore/azerothcore-wotlk/blame/master/src/server/database/Updater/DBUpdater.cpp - I'm a little baffled at the copyright being changed at random. That's not how this all works. This looks like it existed in TrinityCore first, but was copied into AzerothCore and then re-copyrighted. While free software licenses give you a lot of options, that ain't one of them.

Anyway, the updater probably shouldn't assume it's running in the build tree.

**Comment by ChibaPet** [comment]
Hah, publically complaining is the best way to find a silly error in my own process. My post-build migration wasn't moving everything into place. That said, latest build, Brewfest ram racing is still broken. Plus side, I see some long-standing bugs fixed, so that's good.

**Comment by ChibaPet** [comment]
Some [examples](https://ibb.co/S5HMm3q) of [glitching](https://ibb.co/7NSRkqB)...
